refactor(styles): reworked bare padding values with new spacing tokens

GGKapanadze opened the pull request:
## Overview

This PR is a follow-up of #654 and aims to refactor padding values with new spacing tokens for all card components. As mentioned in #637 I don't have access to the foundation file but pretty much all refactoring points I have marked were multiples of 4 so I decided to go with it. 

Here is a refactoring table 

| Card Type      | Component name   | Reworked | Multiple of 4 problem | Design off |
| -------------- | ---------------- | -------- | --------------------- | --------- |
| Text Card      | `_textcard`      | ✅       | ❌                    | ✅        |
| Details Card   | `_detailscard`   | ✅       | ❌                    | ✅        |
| Promo Card     | `_promocard`     | ✅       | ❌                    | ✅        |
| Multilink Card | `_multilinkcard` | ✅       | ❌                    | ✅        |
| Data Card      | `_datacard`      | ✅       | ✅                    | ✅        |
| Stat Card      | `_statcard`      | ✅       | ✅                    | ✅        |
| Fact List Card | `_factlistcard`  | ✅       | ❌                    | ✅        |
| Feature Card   | `_featurecard`   | ✅       | ❌                    | ✅        |
| Card Group     | `_groupcard`     | ❌       | ❌                    | ❌        |

## Rework
Cards marked as reworked have their padding refactored, `_cardgroup` has no padding definition. 

## Multiple of 4 problem
As I mentioned above **almost** all refactoring points were multiples of 4 but there were a few exceptions, that I have refactored 

- `_datacard`: line 93 - **53px  - 52px** (1 pixel off )
- `_datacard`: line 110 - **109px - 108px** (1 pixel off)
- `_statcard`: line 18 - **22px - 24px** (2 pixel off )

## 🔔 Design problems
In the current PR, I have just refactored existing values with a new `spacing token` but I was comparing implementation to Figma and they were a bit off, so I decided to document them for follow-up issues 

Example: 
`Stat card`
> Figma Padding

![image](https://github.com/international-labour-organization/designsystem/assets/36326203/35ccac7d-8fcc-4f97-aaee-5f7804433601)

> Twig Implementation

![image](https://github.com/international-labour-organization/designsystem/assets/36326203/2dda6c1a-37a2-4a36-8704-ed5f9a7c74d7)

Also fixes #569

justintemps commented on packages/styles/scss/components/_datacard.scss:
```diff
@@ -14,7 +14,7 @@
       background: $brand-ilo-light-blue;
       border-bottom: px-to-rem(3px) solid $brand-ilo-ramp-blue;
-      padding: px-to-rem(48px) px-to-rem(52px);
+      padding: map-get($spacing, "spacing-12") map-get($spacing, "spacing-13");
```
Hey @GGKapanadze I think we should conserve these as REM values, what do you think? Would just mean wrapping all of these map-gets() in the px-to-rem() function

GGKapanadze replied:
I'll create a new utility function for spacing like

```
padding: spacing("1")
```
and will wrap implementation inside

justintemps replied:
@GGKapanadze excellent idea

justintemps commented:
@inesdgomes and @beatrizmartinmartins would you guys mind taking a look at this to make sure it's what you're expecting?

https://deploy-preview-674--ilo-ui-twig.netlify.app/?path=/docs/components-cards-card--default

justintemps commented:
Hey @GGKapanadze, let's also take advantage of this to close #569. We should be able to do that simply by addressing this point in @bashlk's comment: https://github.com/international-labour-organization/designsystem/issues/569#issuecomment-1801771284

That should be as simple as bringing the default padding values in line with the ones supplied when it has a "standard" size.

![image](https://github.com/international-labour-organization/designsystem/assets/12401179/a41bb6b5-b962-4555-96e7-994548179099)

GGKapanadze commented:
I have refactored old spacing mixin that wasn't unused. With the new `spacing` helper I have reworked paddings for card components based on Figma [design file ](https://www.figma.com/file/RRxsfLNxjoypKrfJg1EwJf/ILO-Components-Library?type=design&node-id=1433%3A3558&mode=design&t=ptSeCgOPm1HZQXt8-1). 

But there were some nuances

- Cards have inconsistent behavior, for example, the `fact list` card with a wide size was shrinking on a smaller screen but there was no such behavior on Figma. 
- There were extra duplications of the same padding with responsive screens, for example, a `multilink` card
```scss
&__standard {
  --max-width: #{px-to-rem(536px)};

  padding: px-to-rem(48px) px-to-rem(40px);

  @include breakpoint("medium") {
    padding: px-to-rem(48px) px-to-rem(40px);
  }

  @include breakpoint("large") {
    padding: px-to-rem(48px) px-to-rem(40px);
  }
 }
```
----
Here is a table for padding refactoring

| Card Type     | Spacing Reworked | Design Problems |
| ------------- | ---------------- | --------------- |
| Text Card     | ✅               | ❌              |
| Details Card  | ✅               | ❌              |
| Multi Linked  | ❔               | ✅              |
| Data Card     | ✅               | ❌              |
| Stat Card     | ✅               | ❌              |
| FactList Card | ✅               | ❌              |
| Feature Card | ✅               | ❌              |

#### Multilink card
It is a bit inconsistent  from `Figma` to Story book 

- Design sizing: 
	- by Figma props: `wide`, `narrow`, `mobile`?
	- by Title: `wide`, `narrow`
- Story Book sizing: `standard`, `narrow`, `fluid`, `wide`

1. Story book `narrow`  - Figma `mobile`  ??
	-  bottom padding issue
2. Story Book `standard` - Figma `narrow` ?? 

I think we should correctly sync Figma [props ](https://help.figma.com/hc/en-us/articles/5579474826519-Explore-component-properties) to our Story Book or this problem will grow pretty bad. 


### About #569 

First of all, not all sizing matches the Figma file 

For example:
- Standard size in Story Book
![image](https://github.com/international-labour-organization/designsystem/assets/36326203/7de41c58-e016-4e3a-9939-c041a3c4afbf)
- Standard size in Figma
![image](https://github.com/international-labour-organization/designsystem/assets/36326203/056b39d1-06f7-4c08-b208-fac998c7460d)

For default behavior without a specified `size` the card will resize based on screen size, but If the card has a specified size it will get specific padding for all screens without `duplication`, check changed files.

inesdgomes commented:
> @inesdgomes and @beatrizmartinmartins would you guys mind taking a look at this to make sure it's what you're expecting?
> 
> https://deploy-preview-674--ilo-ui-twig.netlify.app/?path=/docs/components-cards-card--default

@justintemps I looked at all the cards and they're looking good to me, though I think the changes may be more evident to Beatriz. @beatrizmartinmartins Could you also take a look please?  https://deploy-preview-674--ilo-ui-twig.netlify.app/?path=/docs/components-cards-card--default Thanks!

bashlk commented:
@justintemps Can we merge this? 3 other PRs depend on it now and it will be more work for the guys the longer this stays unmerged. 

Perhaps we can have a thorough design review after we are done with all the spacing tickets and then fix any of the issues in bulk.
